.pre-commit-config.yaml

```diff
@@ -0,0 +1,121 @@
+# All available hooks: https://pre-commit.com/hooks.html
+# R specific hooks: https://github.com/lorenzwalthert/precommit
+default_install_hook_types:
+  - pre-commit
+  - pre-merge-commit
+
+repos:
+
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    hooks:
+      - id: check-case-conflict
+      - id: check-merge-conflict
+      - id: end-of-file-fixer
+        exclude: '\.Rd'
+      - id: mixed-line-ending
+      - id: trailing-whitespace
+      - id: file-contents-sorter
+        files: '^\.Rbuildignore$'
+
+  - repo: https://github.com/lorenzwalthert/precommit
+    rev: v0.3.2.9019
+    hooks:
+      - id: parsable-R
+        fail_fast: true
+      - id: no-browser-statement
+      - id: no-debug-statement
+      # TODO: enable once a v0.3.2.9020 tag is added
+      # - id: no-print-statement
+      - id: style-files
+        args: [--style_pkg=styler, --style_fun=tidyverse_style]
```

Comment:
Hmm. Generally I am in favor of defining a style. The stricter the better to leave as little ambiguity as possible. However, ideally, we would format all the packages once the style has been settled for. With that in mind, I'll remove this hook for the time being, but I would like to revisit it at some point.

Reply:
Can the extension be configured to not change files in place? (just prevent the commit)

As a context, I've quietly started to like this feature, however, I had to change my workflow to avoid using:

* `git commit --all -m <message>`
    *  in favor of:
        1.  adding the files `git add <file1> ...` or `git add .`
        2. then commit
        3. If styler complains, then I can view the changes styler is enforcing and accept or further change it

Reply:
To see what a specific hook does:
1. head to [its repository](https://github.com/lorenzwalthert/precommit)
1. check [.pre-commit-hooks.yaml](https://github.com/lorenzwalthert/precommit/blob/main/.pre-commit-hooks.yaml#L16-L23)
1. see [what is run in entry](https://github.com/lorenzwalthert/precommit/blob/main/inst/hooks/exported/style-files.R)

In this scenario, there is no `--read-only` flag. However, `style_file` supports `dry = "fail"` so the changes would be minimal. I am already in the process of contributing changes to `{precommit}` and author is open for submissions, so I can eventually implement this too.